src/main/java/org/kuali/kfs/module/ar/document/web/struts/PaymentApplicationAdjustmentAction.java

```diff
@@ -0,0 +1,1291 @@
+                    () -> response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()));
+        return null;
+    }
+
+    public ActionForward addAdHocRouteWorkgroup(
+            final ActionMapping mapping,
+            final ActionForm form,
+            final HttpServletRequest request,
+            final HttpServletResponse response
+    ) throws WorkflowException {
+        final String groupId = extractValueFromDataJson("/group/id", request);
+        final String action = extractValueFromDataJson("/group/action", request);
+
+        if (StringUtils.isBlank(groupId)) {
+            final String msg = String.format("Group id is required");
+            LOG.warn("addAdHocRouteWorkgroup(...) - Returning {}", msg);
+            writeErrorToResponse(response, new ErrorResponse(Response.Status.BAD_REQUEST, msg));
+            return null;
+        }
+
+        if (StringUtils.isBlank(action)) {
+            final String msg = String.format("Action is required");
+            LOG.warn("addAdHocRouteWorkgroup(...) - Returning {}", msg);
+            writeErrorToResponse(response, new ErrorResponse(Response.Status.BAD_REQUEST, msg));
+            return null;
+        }
+
+        final GroupBo group = getBusinessObjectService().findBySinglePrimaryKey(GroupBo.class, groupId);
+        if (ObjectUtils.isNull(group)) {
+            final String msg = String.format("No group found with id=%s", groupId);
+            LOG.warn("addAdHocRouteWorkgroup(...) - Returning {}", msg);
+            writeErrorToResponse(response, new ErrorResponse(Response.Status.NOT_FOUND, msg));
+            return null;
+        }
+
+        final PaymentApplicationAdjustmentForm appaForm = (PaymentApplicationAdjustmentForm) form;
+        loadDocument(appaForm);
+
+        final Document document = appaForm.getDocument();
+        final Person user = getUserFromRequest(request);
+        final AdHocRouteWorkgroup workgroup = new AdHocRouteWorkgroup();
+        workgroup.setRecipientName(group.getName());
+        workgroup.setRecipientNamespaceCode(group.getNamespaceCode());
+
+        if (!getAdHocRoutingService().canSendAdHocRequest(document, workgroup, user)) {
+            final String msg = String.format("No permission to add workgroup with id=%s", groupId);
+            LOG.warn("addAdHocRouteWorkgroup(...) - Returning {}", msg);
+            writeErrorToResponse(response, new ErrorResponse(Response.Status.FORBIDDEN, msg));
+            return null;
+        }
+
+        final AdHocRouteWorkgroupResponse workgroupResponse = appaForm.createAdHocRouteWorkgroupResponse(group, action);
+        serializeToJsonSafely(workgroupResponse)
+                .ifPresentOrElse(
+                    json -> writeJsonToResponse(response, Response.Status.OK, json),
+                    () -> response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()));
+        return null;
+    }
+    
+    /*
+     * CU Customization KFSPTS-21426
```

Comment:
should this be marked as a backport?

Reply:
similar, KFSPTS-21426 - backport FINP 7302

Reply:
sounds good